src/mongoose_rabbit_worker.erl

```diff
@@ -117,64 +113,28 @@ terminate(_Reason, #state{connection = Connection, channel = Channel,
 %%% Internal functions
 %%%===================================================================
 
--spec declare_exchange(Channel :: pid(), Name :: binary(), Type :: binary()) ->
-  term().
-declare_exchange(Channel, Name, Type) ->
-    amqp_channel:call(Channel, #'exchange.declare'{exchange = Name,
-                                                   type = Type}).
-
--spec delete_exchange(Channel :: pid(), Exchange :: binary()) -> term().
-delete_exchange(Channel, Exchange) ->
-    amqp_channel:call(Channel, #'exchange.delete'{exchange = Exchange}).
-
--spec publish_status(map(), Opts :: publish_opts()) -> term().
-publish_status(#{user_jid := {User, Host, _} = JID, exchange := Exchange,
-                 status := Status}, Opts) ->
-    RoutingKey = jid:to_binary({User, Host}),
-    Message = make_presence_msg(JID, Status),
-    publish(Exchange, RoutingKey, Message, Opts#state{host = Host}).
-
--spec publish_chat_msg_sent(EventData :: map(), Opts :: publish_opts()) -> ok.
-publish_chat_msg_sent(EventData = #{from_jid := From, topic := Topic}, Opts) ->
-    RoutingKey = user_topic_routing_key(From, Topic),
-    publish_chat_msg_event(RoutingKey, EventData, Opts).
-
--spec publish_chat_msg_received(EventData :: map(), Opts :: publish_opts()) -> ok.
-publish_chat_msg_received(EventData = #{to_jid := To, topic := Topic}, Opts) ->
-    RoutingKey = user_topic_routing_key(To, Topic),
-    publish_chat_msg_event(RoutingKey, EventData, Opts).
-
--spec publish_groupchat_msg_sent(EventData :: map(), Opts :: publish_opts()) -> ok.
-publish_groupchat_msg_sent(EventData = #{from_jid := From, topic := Topic},
-                           Opts) ->
-    RoutingKey = user_topic_routing_key(From, Topic),
-    publish_chat_msg_event(RoutingKey, EventData, Opts).
-
--spec publish_groupchat_msg_recv(EventData :: map(), Opts :: publish_opts()) -> ok.
-publish_groupchat_msg_recv(EventData = #{to_jid := To, topic := Topic}, Opts) ->
-    RoutingKey = user_topic_routing_key(To, Topic),
-    publish_chat_msg_event(RoutingKey, EventData, Opts).
-
--spec publish_chat_msg_event(RoutingKey :: binary(), map(), Opts :: publish_opts()) ->
-    term().
-publish_chat_msg_event(RoutingKey, #{from_jid := From,
-                                     to_jid := To,
-                                     msg := UserMsg,
-                                     exchange := Exchange,
-                                     host := Host}, Opts) ->
-    Message = make_chat_msg(From, To, UserMsg),
-    publish(Exchange, RoutingKey, Message, Opts#state{host = Host}).
+-spec handle_amqp_call(Method :: mongoose_amqp:method(), worker_opts()) ->
+    {ok, term()} | {error, term(), term()}.
+handle_amqp_call(Method, #state{channel = Channel}) ->
+    try amqp_channel:call(Channel, Method) of
```

Comment:
I am not sure, if catching is ok here. If exception is thrown, the channel is closed. If we catch it, the process is still participating in the pool as a worker and accepts the requests.  I think we should try to reopen connection/channel to avoid loosing messages from message queue and then crash without catching any exceptions if it fails.